app-backend/datamodel/src/main/scala/AnnotationLabelClassGroup.scala

```diff
@@ -0,0 +1,54 @@
+package com.rasterfoundry.datamodel
+
+import io.circe.generic.semiauto._
+import io.circe._
+
+import java.util.UUID
+
+final case class AnnotationLabelClassGroup(
+    id: UUID,
+    name: String,
+    annotationProjectId: UUID,
+    index: Int
+) {
+  def withLabelClasses(
+      classes: List[AnnotationLabelClass]
+  ): AnnotationLabelClassGroup.WithRelated =
+    AnnotationLabelClassGroup.WithRelated(
+      id,
+      name,
+      annotationProjectId,
+      index,
+      classes
+    )
+}
+
+object AnnotationLabelClassGroup {
+  implicit val encAnnotationLabelClassGroup
+    : Encoder[AnnotationLabelClassGroup] = deriveEncoder
+  implicit val decAnnotationLabelClassGroup
+    : Decoder[AnnotationLabelClassGroup] = deriveDecoder
+
+  final case class Create(
+      name: String,
+      index: Option[Int],
+      classes: List[AnnotationLabelClass.Create]
+  )
+
+  object Create {
+    implicit val decAnnotationLabelClassGroupCreate
+      : Decoder[AnnotationLabelClassGroup.Create] = deriveDecoder
+  }
+
+  final case class WithRelated(
```

Comment:
Yeah it's not a great name. I used it because it's what we used on scenes, images, user group roles, and tool runs. Since in this case there's just the one type of related thing I'll make the change. The name gets kind of screwy for `AnnotationProject.WithRelated` though (`WithTileLayersAndLabelClassGroups`?) so I'm planning to leave that one.